### PR TITLE
Implement loadNeuron() for map and bin reports

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: brion/compartmentReport.cpp [hunk 1]
```diff
@@ -120,6 +120,23 @@ floatsPtr CompartmentReport::loadFrame( const float timestamp ) const
     return _impl->plugin->loadFrame( timestamp );
 }
 
+size_t CompartmentReport::getNeuronSize( const uint32_t gid ) const
+{
+    const auto& gids = getGIDs();
+    const size_t index = std::distance( gids.begin(), gids.find( gid ));
+    if( index >= gids.size( ))
+        LBTHROW( std::runtime_error( "Gid " + std::to_string( gid ) +
+                                     " not in report" ));
```
Comment (rdumusc) [inline brion/compartmentReport.cpp]: could even go to:
size_t _getIndex( const uint32_t gid ) const { ... }

File: brion/compartmentReport.h [hunk 2]
```diff
@@ -122,17 +122,32 @@ class CompartmentReport : public boost::noncopyable
...
      * @param timestamp the time stamp of interest
-     * @return the report values if found at timestamp, 0 otherwise
+     * @return the report values if found at timestamp, nullptr otherwise
      * @version 1.0
      */
     BRION_API floatsPtr loadFrame( float timestamp ) const;
 
+    /**
+     * @return the number of values of the given neuron report.
+     * @throw std::runtime_error if gid is not mapped.
```
Comment (rdumusc) [inline brion/compartmentReport.h]: missing param

File: brion/plugin/compartmentReportMap.cpp [hunk 3]
```diff
@@ -394,6 +394,45 @@ floatsPtr CompartmentReportMap::loadFrame( const float time ) const
...
+floatsPtr CompartmentReportMap::loadNeuron( const uint32_t gid ) const
+{
+    if( !_readable )
+        return floatsPtr();
+
+    const size_t index = std::distance( _gids.begin(), _gids.find( gid ));
+    if( index >= _gids.size( ))
+        return floatsPtr();
+
+    const std::string& scope = _uri + std::to_string( gid ) + "_";
```
Comment (rdumusc) [inline brion/plugin/compartmentReportMap.cpp]: style: drop the & for local variable which is not actually a ref? const std::string or const auto

File: brion/plugin/compartmentReportMap.cpp [hunk 4]
```diff
@@ -416,23 +455,27 @@ floatsPtr CompartmentReportMap::loadFrame( const float time ) const
...
             std::free( data );
-            ++taken;
         };
 
         store.takeValues( subKeys, takeValue );
     }
 
-    if( size_t( taken ) == _gids.size( ))
-        return buffer;
+    if( size_t( taken ) == keys.size() )
```
Comment (rdumusc) [inline brion/plugin/compartmentReportMap.cpp]: style: spaces ))

File: doc/Changelog.md [hunk 5]
```diff
@@ -3,6 +3,8 @@ Changelog {#Changelog}
 
 # git master
 
+* [122](https://github.com/BlueBrain/Brion/pull/122):
+  Add loadNeuron for binary and map compartment reports
```
Comment (rdumusc) [inline doc/Changelog.md]: -> Add the capacity to load individual neurons [...]

File: apps/compartmentConverter.cpp [hunk 6]
```diff
@@ -79,6 +79,8 @@ int main( const int argc, char** argv )
         ( "output,o", po::value< std::string >(), "Output report URI" )
         ( "maxFrames,m", po::value< size_t >(),
           "Convert at most the given number of frames" )
+        ( "gids,g", po::value< std::vector< uint32_t >>()->multitoken(),
+          "List of whitespace-separated sub-GIDs to convert" )
```
Comment (tribal-tec) [inline apps/compartmentConverter.cpp]: rm sub-?

File: brion/compartmentReport.cpp (hunk 1 above)
Comment (eile) [inline brion/compartmentReport.cpp]: done

File: brion/compartmentReport.h (hunk 2 above)
Comment (eile) [inline brion/compartmentReport.h]: done

File: brion/plugin/compartmentReportMap.cpp (hunk 3 above)
Comment (eile) [inline brion/plugin/compartmentReportMap.cpp]: Will do in #123 to avoid conflicts.

File: brion/plugin/compartmentReportMap.cpp (hunk 4 above)
Comment (eile) [inline brion/plugin/compartmentReportMap.cpp]: done

File: doc/Changelog.md (hunk 5 above)
Comment (eile) [inline doc/Changelog.md]: done

File: apps/compartmentConverter.cpp (hunk 6 above)
Comment (eile) [inline apps/compartmentConverter.cpp]: done

**Pull request merged**